NEWS

```diff
@@ -1,5 +1,185 @@
+
+  - It now includes the backup type in the JSON output, which can be either
+    `rsync` for backups taken with rsync, `full` or `incremental` for backups
+    taken with `pg_basebackup`, or `snapshot` for cloud snapshots. When printing
+    to the console the backup type is represented by the corresponding labels
+    `R`, `F`, `I` or `S`.
+
+  - Remove tablespaces information from the output. That was bloating the
+    output. Tablespaces information can still be taken from `barman show-backup`
+    output.
+
+- Always set a timestamp with a time zone when configuring
+  `recovery_target_time` through `barman recover`. Previously, if no time zone
+  was explicitly set through `--target-time`, Barman would configure
+  `recovery_target_time` without a time zone in Postgres. Without a time zone,
+  Postgres would assume whatever is configured through `timezone` GUC in
+  Postgres. From now on Barman will issue a warning and configure
+  `recovery_target_time` with the time zone of the Barman host, if no time zone
+  is set by the user through `--target-time` option.
+
+- When recovering a backup with “no get wal” approach and `--target-lsn` is set,
+  copy only the WAL files required to reach the configured target. Previously
+  Barman was copying all the WAL files from its archive to Postgres.
+
+- When recovering a backup with “no get wal” approach and `--target-immediate`
+  is set, copy only the WAL files required to reach the consistent point.
+  Previously Barman was copying all the WAL files from its archive to Postgres.
+
+- `barman-wal-restore` now moves WALs from the spool directory to `pg_wal`
+  instead of copying them. This can improve performance if the spool directory
+  and the `pg_wal` directory are in the same partition of the filesystem.
+
+- `barman check-backup` now shows the reason why a backup was marked as `FAILED`
+  in the output and logs. Previously for a user to know why the backup was
+  marked as `FAILED`, they would need to run `barman show-backup` command.
+
+- Add configuration option `aws_await_snapshots_timeout` and the corresponding
+  `--aws-await-snapshots-timeout` command-line option on `barman-cloud-backup`.
+  This specifies the timeout in seconds to wait for snapshot backups to reach
+  the completed state.
+
+- Add a keep-alive mechanism to rsync-based backups. Previously the Postgres
+  session created by Barman to run `pg_backup_start` and `pg_backup_stop` would
+  be idle for how long the base backup copy could take. That could lead to a
+  firewall or router dropping the connection because it was idle for a long
+  time. The keep-alive mechanism keeps sending heartbeat queries to Postgres
+  through that connection, thus reducing the likelihood of a connection
+  droppage. The interval between heartbeats can be controlled through the new
+  configuration option `keepalive_interval` and through the corresponding CLI
+  option `--keepalive-interval` of the `barman backup` command.
```

Comment:
```suggestion
- Add a keep-alive mechanism to rsync-based backups. Previously the Postgres
  session created by Barman to run `pg_backup_start()` and `pg_backup_stop()` would
  stay idle for as long as the base backup copy would take. That could lead to a
  firewall or router dropping the connection because it was idle for a long
  time. The keep-alive mechanism sends heartbeat queries to Postgres
  through that connection, thus reducing the likelihood of a connection
  getting dropped. The interval between heartbeats can be controlled through the new
  configuration option `keepalive_interval` and the corresponding CLI
  option `--keepalive-interval` of the `barman backup` command.
```